src/ResourceManager/Common/Commands.ResourceManager.Common/ResourceGroupCompleter.cs

```diff
@@ -0,0 +1,134 @@
+                        //var client = AzureSession.Instance.ClientFactory.CreateArmClient<ResourceManagementClient>(context, AzureEnvironment.Endpoint.ResourceManager);
+                        client.SubscriptionId = context.Subscription.Id;
+                        var resourceGroups = client.ResourceGroups.ListAsync();
+                        if (resourceGroups.Wait(TimeSpan.FromSeconds(5)))
+                        {
+                            if (resourceGroups.Result != null)
+                            {
+                                var resourceGroupList = resourceGroups.Result.ToList();
+                                foreach (ResourceGroup resourceGroup in resourceGroupList)
+                                {
+                                    _resourceGroupNames.Add(resourceGroup.Name);
+                                }
+                            }
+
+                            else
+                            {
+#if DEBUG
+                                throw new Exception("Result from client.ResourceGroups is null");
+#endif
+                            }
+                        }
+
+                        else
+                        {
+#if DEBUG
+                            throw new Exception("client.ResourceGroups call timed out");
+#endif
+                        }
+                    }
+
+                    catch (Exception ex)
+                    {
+#if DEBUG
+                        throw ex;
+#endif
+                    }
+                }
+
+                return _resourceGroupNames;
+            }
+        }
+
+        /// <summary>
+        /// This class will provide a list of resource groups that are available to the user (with default resource group first if it exists). This will then be available to the user to tab through.
+        /// Example: [Parameter(ParameterSetName = ListByNameInTenantParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = false), ResourceGroupCompleter()]
+        /// </summary>
+        /// <param name="resourceTypes"></param>
+        public ResourceGroupCompleterAttribute() : base(CreateScriptBlock())
+        {
+        }
+
+        public static string[] GetResourceGroups()
+        {
+            IAzureContext context = AzureRmProfileProvider.Instance.Profile.DefaultContext;
+            var resourceGroupNamesCopy = ResourceGroupNames;
+            if (context.ExtendedProperties.ContainsKey(Resources.DefaultResourceGroupKey))
+            {
+                return GetResourceGroups(resourceGroupNamesCopy, context.ExtendedProperties[Resources.DefaultResourceGroupKey]);
+            }
+            return GetResourceGroups(resourceGroupNamesCopy, null);
+        }
+
+        public static string[] GetResourceGroups(IList<string> resourceGroupNames, string defaultResourceGroup)
+        {
+            if (defaultResourceGroup != null)
+            {
+                if (resourceGroupNames.Contains(defaultResourceGroup))
```

Comment:
What if the default resource group is on page 2 of the results?  Seems like we should always include the defautl resource group at the beginning

Reply:
If the user types out part of a name not including the default resource group, I think it makes sense not to include the default resource group.  Let me know if I should include it.

Reply:
Agreed, but it still could be in the second page of results

Reply:
Still waiting to hear back from the Resources team about if it is possible to query Resource Groups by name